### PR TITLE
grass.jupyter: fix resolution handling for InteractiveMap

### PR DESCRIPTION
**Pull request by petrasovaa**
Should address #2369. Now when use_region=True or saved_region, it should reproject the map with the correct resolution. Note that in these cases the resolution is not estimated by r.proj but is estimated by reprojecting the extent and using the same ncols/nrows as in the source region, which might not be the most precise way.

### CODE REVIEW

**Comment by neteler** [comment]
I have tested this with my larger dataset:

image

Then, when displaying it. I get:

image
image

Checking in parallel what is run underneath, with `ps aux | grep proj` I see:

`r.proj input=opennrw_lidar_bonn_p95_50cm_DOM@bonn_lidar output=opennrw_lidar_bonn_p95_50cm_DOM_bonn_lidar mapset=bonn_lidar location=openNRW_epsg25832 dbase=/home/mneteler/grassdata resolution=0.8094142783201477`

What I notice is that I set the computational region to `res=2m` while `r.proj` uses `resolution=0.809` which is resulting in way too many pixels. Possibly my mistake?

**Comment by neteler** [comment]
> Try use_region=True

Ok, this then changes the resolution (not sure how to inspect the bbox):

`r.proj input=opennrw_lidar_bonn_p95_50cm_DOM@bonn_lidar output=opennrw_lidar_bonn_p95_50cm_DOM_bonn_lidar mapset=bonn_lidar location=openNRW_epsg25832 dbase=/home/mneteler/grassdata resolution=3.21940318`

And it helps:

image

The question is now what should be the default: `use_region=True` or not, as currently.
Perhaps catch the error above and in case suggest to the user to add `use_region=True`?

File: python/grass/jupyter/interactivemap.py [hunk 1]
```diff
@@ -165,6 +165,16 @@ def __init__(
         folium documentation:
         https://python-visualization.github.io/folium/modules.html
 
+        Raster and vector data are always reprojected to pseudomercator.
+        With use_region=True or saved_region=myregion region extent
```
Comment (veroandreo) [inline python/grass/jupyter/interactivemap.py:169]:
```suggestion
        With use_region=True or saved_region=myregion, the region extent
```

File: python/grass/jupyter/interactivemap.py [hunk 2]
```diff
@@ -165,6 +165,16 @@ def __init__(
...
+        is reprojected and the number of rows and columns of that region
+        is kept the same. This region is then used for reprojection.
+        By default use_region is False, resulting in
```
Comment (veroandreo) [inline python/grass/jupyter/interactivemap.py:172]:
```suggestion
        By default, use_region is False, which results in the
```

File: python/grass/jupyter/interactivemap.py [hunk 3]
```diff
@@ -165,6 +165,16 @@ def __init__(
...
+        reprojecting entire raster in its native resolution,
+        the reprojected resolution is estimated with r.proj.
```
Comment (veroandreo) [inline python/grass/jupyter/interactivemap.py:174]:
```suggestion
        reprojection of the entire raster in its native resolution.
        The reprojected resolution is estimated with r.proj.
```

**Review by veroandreo — COMMENTED** [review]
I added grammar suggestions in the documentation paragraph :)

**Review by wenzeslaus — APPROVED** [review]
The code and behavior make sense to me.

**Pull request merged**